extensions/amp-web-push/0.1/permission-dialog.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import {tryDecodeUriComponent} from '../../../src/url.js';
+import {WindowMessenger} from './window-messenger';
+
+/**
+ * @fileoverview
+ * The script for the web push notification permission dialog. This script will
+ * eventually live on the publisher's origin. It shows the notification prompt
+ * and forwards results to the AMP page.
+ */
+class AmpWebPushPermissionDialog {
+  constructor(options) {
+    if (options && options.debug) {
+      // Debug enables verbose logging for this page and the window and worker
+      // messengers
+      this.debug = true;
```

Comment:
Fixed.

Changed the `if { .. }` block setting the boolean flag to just `this.debug_ = options && options.debug;` with the flag being private.